spec.bs

```diff
@@ -935,14 +935,24 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
         [=serializing a JavaScript value to a JSON string=], given |result|.
     * To handle an error, set |auctionConfig|'s [=auction config/seller signals=] to failure.
 1. If |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] [=map/exists=]:
-  1. Set |auctionConfig|'s [=auction config/expects direct from seller signals header ad slot=] to true.
   1. [=Handle an input promise in configuration=] given |auctionConfig| and
     |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"]:
+    
+    Note: The JavaScript code calling {{Navigator/runAdAuction()}} is responsible for not resolving
+      |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] until a
+      {{WindowOrWorkerGlobalScope/fetch()}} call with the `{adAuctionHeaders: true}` option returns
+      a response, or alternatively, not calling {{Navigator/runAdAuction()}} until the

```

Comment:
yes, that's what I meant, but I'm not confident that it's the best wording.

Reply:
I think this works, but maybe @domfarolino has thoughts?

Reply:
Is the `directFromSellerSignalsHeaderAdSlot` here supposed to contain any data relevant to the `fetch(..., {adAuctionDataThingThatCalebAdded: true})` request?

Reply:
Two points:

1. This discussion seems to obsolete https://github.com/WICG/turtledove/pull/774#discussion_r1340492472, right? Do you think we can mark that one as resolved?
2. Regarding https://github.com/WICG/turtledove/pull/774/files#r1340641527 specifically, I think we should do two things:
    - Use language like "is responsible for resolving directFromSellerSignalsHeaderAdSlot *only* when the Promise returned from an associated fetch() that was made with the `{adAuctionHeaders: true}` option resolves or rejects."
    - Describe what happens if ad tech _doesn't_ do this. I think the result is the "race condition" that I brought up before, where the worklet can run _without_ the data that it needs(?) because it is still in-flight from the server. I think describing this consequence would be good.

Reply:
1. yes, resovled that comment.
2. Done. Updated the `Note` with the suggestions.